intelliserver/api/parser/index.js

```diff
@@ -0,0 +1,44 @@
+const express = require('express');
+const multer = require('multer');
+
+const pdf = require('pdf-parse');
+const mammoth = require('mammoth');
+
+const router = express.Router();
+const upload = multer();
+
+router.post('/pdf_to_text', upload.single('pdf'), (req, res) => {
+    if (!req.file) {
+        return res.status(400).json({ error: 'PDF document not provided' });
+    }
+    const { buffer } = req.file;
+
+    pdf(buffer)
+        .then(data => {
+            const text = data.text;
```

Comment:
Return the text of each page in a dictionary.

The response should match the below format
```
{
  "status": "success",
  "data": {
        "total_pages": 5,    
        "pages": [
		{
                "page_number": 1,
                "text": "This is the text content of the first page"
                },
                {
                "page_number": 2,
                "text": "This is the text content of the second page"
                },
                ....
      ]
  }
}
```